merlin/dataloader/loader_base.py

```diff
@@ -569,22 +581,34 @@ def _handle_tensors(self, tensors, tensor_names):
                 tensor, lists = tensor
 
             names = [i for i in names if i not in lists]
+            if not self.lists_as_tuple:
+                names_list = list(lists.keys())
```

Comment:
There's a utility function for this in Core, which might be handy here: https://github.com/NVIDIA-Merlin/core/blob/main/merlin/dag/utils.py#L18-L41

Reply:
Nice I will update it